storage/database/db_manager.go

```diff
@@ -640,6 +638,9 @@ func (dbm *databaseManager) CreateMigrationDBAndSetStatus(blockNum uint64) error
 	// Create a new database for migration process.
 	newDB, newDBDir := newStateTrieMigrationDB(dbm.config, blockNum)
 
+	dbm.lockInMigration.Lock()
+	defer dbm.lockInMigration.Unlock()
```

Comment:
```suggestion
	// lock to prevent from a conflict of reading state DB and changing state DB
	dbm.lockInMigration.Lock()
	defer dbm.lockInMigration.Unlock()
```
To be sure, how about adding a comment like the code below?